app/assets/javascripts/heatmap.js

```diff
@@ -0,0 +1,116 @@
+/* global CalHeatmap, CalendarLabel, Tooltip */
+document.addEventListener("DOMContentLoaded", () => {
+  const heatmapElement = document.querySelector("#cal-heatmap");
+
+  if (!heatmapElement) {
+    console.warn("Heatmap element not found in the DOM.");
+    return;
+  }
+
+
+  const heatmapData = heatmapElement.dataset.heatmap ? JSON.parse(heatmapElement.dataset.heatmap) : [];
+
+  try {
+    // eslint-disable-line no-undef
+    const cal = new CalHeatmap();
+    const applicationData = $("head").data();
+    const locale = applicationData.locale;
+    const theme = getTheme();
```

Comment:
We can watch for color theme with 
https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/change_event

Reply:
Refactored the code accordingly so it takes into account if the theme of the browser is changed. Thank you for pointing this out